rust/lance/src/dataset.rs

```diff
@@ -974,119 +625,22 @@ impl Dataset {
             Ok,
         )?;
 
-        let (object_store, base, commit_handler) = Self::params_from_uri(
-            base_uri,
-            &commit_handler,
-            &store_params,
-            object_store_registry.clone(),
-        )
-        .await?;
+        let transaction = Transaction::new(read_version, operation, blobs_op, None);
 
-        // Test if the dataset exists
-        let dataset_exists = match commit_handler
-            .resolve_latest_version(&base, &object_store)
-            .await
-        {
-            Ok(_) => true,
-            Err(Error::NotFound { .. }) => false,
-            Err(e) => return Err(e),
-        };
+        let mut builder = CommitBuilder::new(base_uri)
+            .with_object_store_registry(object_store_registry)
+            .enable_v2_manifest_paths(enable_v2_manifest_paths)
+            .with_detached(detached);
```

Comment:
It seems a little strange to have a builder pattern here but not expose it to the API (e.g. have `do_commit` return `CommitBuilder`.  Is the goal simply to maintain the API for now for backwards compatibility?

Reply:
The builders are public. I just didn't want to disrupt the API surface too much. I figure we'll deprecate the old methods later in favor of the builders.